### PR TITLE
Add individual padding attributes for the cell and message labels

### PR DESCRIPTION
**Pull request by mdescalzo**
What does this implement/fix? Explain your changes.
---------------------------------------------------
- Adds `cellTopLabelPadding`, `cellBottomLabelPadding`, `messageTopLabelPadding`, & `messageBottomLabelPadding` attributes to `MessagesCollectionViewLayoutAttributes`
- Adds the above attributes to layout calculations in `MessageSizeCalculator`
- Adds `cellTopLabelPadding`, `cellBottomLabelPadding`, `messageTopLabelPadding`, & `messageBottomLabelPadding` methods to the `MessageLayoutDelegate` protocol with defaults which return `.zero` insets.

This allows manipulation of the label frames which can be desirable when adding borders or backgrounds to these labels.

The default `.zero` insets means this should not break/change existing layout behavior.

Does this close any currently open issues?
------------------------------------------
This is, in part, to address issues discussed in #1107 

Any relevant logs, error output, etc?
-------------------------------------
nope

Any other comments?
-------------------
nope

Where has this been tested?
---------------------------
**Devices/Simulators:** 
iPad sim, iPhone Xs, iPad mini

**iOS Version:** …
12.2

**Swift Version:** …
4.2 & 5

**MessageKit Version:**
3.0.0

### CODE REVIEW

File: Sources/Protocols/MessagesLayoutDelegate.swift [hunk 1]
```diff
@@ -145,18 +185,33 @@ public extension MessagesLayoutDelegate {
         return 0
     }
     
+    func cellTopLabelPadding(for message: MessageType, at indexPath: IndexPath) -> UIEdgeInsets {
```
Comment (nathantannar4) [inline Sources/Protocols/MessagesLayoutDelegate.swift:188]: We cannot set these properties through a delegate method. See #1107

File: Sources/Layout/MessagesCollectionViewLayoutAttributes.swift [hunk 2]
```diff
@@ -40,15 +40,19 @@ open class MessagesCollectionViewLayoutAttributes: UICollectionViewLayoutAttribu
 
     public var cellTopLabelAlignment = LabelAlignment(textAlignment: .center, textInsets: .zero)
     public var cellTopLabelSize: CGSize = .zero
+    public var cellTopLabelPadding: UIEdgeInsets = .zero
```
Comment (nathantannar4) [inline Sources/Layout/MessagesCollectionViewLayoutAttributes.swift:43]: This is where we want the padding properties, so its confusing how there is a mixture of this and delegates. We should also have a convenience setter in `MessagesCollectionViewFlowLayout` to set this as we do with the other properties

**Review by nathantannar4 — CHANGES_REQUESTED** [review]
I'm happy to add padding properties, but it should follow the same convention we have for the other layout attributes.

**Comment by mdescalzo** [comment]
I'm pulling this back for a rework.  Thank you very much for the input.

**Comment by nathantannar4** [comment]
Look forward to the new version!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request closed without merging**